packages/toolpad-app/src/toolpadDataSources/rest/client.tsx

```diff
@@ -223,35 +212,14 @@ function ResolvedPreview({ preview }: ResolvedPreviewProps): React.ReactElement
 
   const { untransformedData } = preview;
 
-  if (!untransformedData || isEmpty(untransformedData)) {
+  if (untransformedData === undefined) {
     return (
```

Comment:
Just an idea, but would it make sense to both show the `JsonView` with `undefined` **and** the Alert? Basically we'd always show the exact JSON output, but in some cases we show an extra warning alongside of it.

Reply:
You mean this alert? 👇 (sorry for code screenshot 🙈 )
<img width="591" alt="image" src="https://user-images.githubusercontent.com/437214/196214899-7dcf6fe7-452a-4f05-84d0-76ed661c2f28.png">

Would you like to display that only when `undefined` is returned or some other conditions as well?

Reply:
IMO all suggestions are easily doable right now. I could make the changes and just render same text that I've used before without auto tab switching. I can also add a link to a tab on `transform` word so user could quickly jump into that if needed